nconnect.go

```diff
@@ -371,37 +391,44 @@ func (nc *nconnect) StartClient() error {
 	proxyHost := proxyAddr.IP.String()
 	proxyPort := uint16(proxyAddr.Port)
 
-	var from, to []string
-	for _, remote := range remoteTunnelAddr {
-		port, err := util.GetFreePort()
-		if err != nil {
-			return err
-		}
+	if len(remoteTunnelAddr) > 0 {
+		var from, to []string
+		for _, remote := range remoteTunnelAddr {
+			port, err := util.GetFreePort(0)
+			if err != nil {
+				return err
+			}
 
-		ssAddr := "127.0.0.1:" + strconv.Itoa(port)
-		from = append(from, ssAddr)
-		to = append(to, remote)
+			ssAddr := "127.0.0.1:" + strconv.Itoa(port)
+			from = append(from, ssAddr)
+			to = append(to, remote)
 
-		if remoteInfo, ok := nc.remoteInfoByTunnel[remote]; ok {
-			for _, addr := range remoteInfo.LocalIP.Ipv4 {
-				nc.ssConfig.TargetToClient[addr] = ssAddr
+			if remoteInfo, ok := nc.remoteInfoByTunnel[remote]; ok {
+				for _, addr := range remoteInfo.LocalIP.Ipv4 {
+					nc.ssClientConfig.TargetToClient[addr] = ssAddr
+				}
 			}
 		}
-	}
-	tunnels, err := tunnel.NewTunnels(nc.account, nc.opts.Identifier, from, to, nc.opts.Tuna, nc.tunnelConfig)
-	if err != nil {
-		return err
-	}
-	nc.tunnels = tunnels
 
-	nc.ssConfig.Socks = nc.opts.LocalSocksAddr
-	nc.ssConfig.Client = from[0]
-	nc.ssConfig.DefaultClient = from[0] // the first config is the default client
+		identifier := config.RandomIdentifier()
+		tunnels, err := tunnel.NewTunnels(nc.account, identifier, from, to, nc.opts.Tuna, nc.tunnelConfig)
+		if err != nil {
+			return err
+		}
+		nc.tunnels = tunnels
+
+		nc.ssClientConfig.Client = from[0]
+		nc.ssClientConfig.DefaultClient = from[0] // the first config is the default client
+	} else {
+		nc.ssClientConfig.Client = "127.0.0.1:0"
+		nc.ssClientConfig.DefaultClient = ""
+	}
+	nc.ssClientConfig.Socks = nc.opts.LocalSocksAddr
 
 	log.Println("Client socks proxy listen address:", nc.opts.LocalSocksAddr)
 
 	if nc.opts.Tun || nc.opts.VPN {
-		tunDevice, err := arch.OpenTunDevice(nc.opts.TunName, nc.opts.TunAddr, nc.opts.TunGateway, nc.opts.TunMask, nc.opts.TunDNS, true)
+		tunDevice, err := arch.OpenTunDevice(nc.opts.TunName, nc.opts.TunAddr, nc.opts.TunGateway, nc.opts.TunMask, nc.opts.TunDNS, false)
```

Comment:
FYI persist option only works on Linux

Reply:
I see. let me restore it.

Reply:
I mean you can still use it if needed, but you need to make sure the changes will work on all platforms we support

Reply:
Yes, thanks for the advice.